notifiers/github-app/pkg/controller/gh_metadata.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The Tekton Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// Extract the repo and owner information from the Annotations
+func getRepoMetadata(tr *v1beta1.TaskRun) (string, string, error) {
+	var owner string
+	var repo string
+	// The repo-url Annotation takes precedence
+	// over separate owner, repo Annotations
+	repoUrl := tr.Annotations[key("repo-url")]
+	if repoUrl != "" {
+		s := strings.Split(repoUrl, "/")
```

Comment:
Instead of relying on string splitting you might want to consider using Go's `net/url` package to parse URLs in a more deterministic fashion. https://gobyexample.com/url-parsing